universum/configuration_support.py

```diff
@@ -7,15 +8,131 @@
 __all__ = [
     "Step",
     "Configuration",
-    "Variations",
     "set_project_root",
     "get_project_root"
 ]
 
 
 class Step:
     """
-    Step is a collection of configurable project-specific data needed for Universum operations.
+    Step is a collection of configurable project-specific data entries needed for Universum functionality and passed to
+    a single Universum launch. Individual build steps are collected in a :class:`Configuration` object, which supports
```

Comment:
Individual build steps are collected in a `Configuration` object, that allows to create linear and nested step structures.
_(I don't think this is a right place to explain 'combining and chaining' anyway)_